src/Modal/index.js

```diff
@@ -1,20 +1,36 @@
-import React from 'react'
+import React, { useEffect } from 'react'
 import {default as PT} from 'prop-types'
 import cx from 'classnames'
 import Icon from '../Icon'
 
-const Modal = ({ extraClass, children, onClose, ...props }) => {
+const Overlay = ({ children, open, ...props }) => open ?
+  <div className="ola_modal-overlay" {...props}>
+    {children}
+  </div> : null
+
+const Modal = ({ extraClass, children, closeWhenClickOut = true, closeWhenPressEsc = true, onClose, ...props }) => {
+    
+  useEffect(() => closeWhenPressEsc && document.body.addEventListener('keydown', e => e.keyCode === 27 && onClose()), [])
+
   return (
-    <section className={cx('ola_modal', extraClass)} {...props}>
-      <button className="ola_modal-close ola_buttonIcon" onClick={() => onClose()}>
-        <Icon name="close" />
-      </button>
-      {children}
-    </section>
+    <Overlay onClick={closeWhenClickOut && onClose} {...props}>
+      <dialog onClick={e => e.stopPropagation()} className={cx('ola_modal', extraClass)} {...props}>
+        <button className="ola_modal-close ola_buttonIcon" onClick={onClose}>
+          <Icon name="close" />
+        </button>
+        {children}
+      </dialog>
+    </Overlay>
   )
 }
 
 Modal.propTypes = {
+  /** Open or close Modal */
+  open: PT.bool,
+  /** Close Modal when click out. Default is true */
+  closeWhenClickOut: PT.bool,
```

Comment:
`closeWhenClickOut` and `closeWhenPressEsc` props should always be true. Any case where this is not so? I think it is better to remove these props